### PR TITLE
Add PkgProj for System.Data.DataSetExtensions

### PR DESCRIPTION
**Pull request by nchikanov**
Updated necessary descriptions and packaging for System.Data.DataSetExtensions (part of #22803)

### CODE REVIEW

File: pkg/descriptions.json [hunk 1]
```diff
@@ -1,2074 +1,2081 @@
...
-      "System.Security.Cryptography.Aes",
-      "System.Security.Cryptography.RSA",
-      "System.Security.Cryptography.RSAParameters",
-      "System.Security.Cryptography.HMACSHA1",
-      "System.Security.Cryptography.SHA256",
-      "System.Security.Cryptography.SHA1",
-      "System.Security.Cryptography.SHA512",
-      "System.Security.Cryptography.SHA384",
-      "System.Security.Cryptography.HMACSHA256",
-      "System.Security.Cryptogr
```
Comment (weshaggard) [inline pkg/descriptions.json]: Did you convert this file to having tabs? Or did you just resize all the tabs? In general we try to match the formatting of the file as opposed to resetting the entire thing. As it stands it is hard to see what you actually changed in this diff.

File: src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup>
+    <ProjectReference Include="..\ref\System.Data.DataSetExtensions.csproj">
```
Comment (weshaggard) [inline src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj]: You are going to also need to add some placeholder information for net45. I think you should pretty much copy https://github.com/dotnet/corefx/blob/master/src/System.DirectoryServices/pkg/System.DirectoryServices.pkgproj#L10.

File: pkg/descriptions.json (hunk 1 above)
Comment (nchikanov) [inline pkg/descriptions.json]: the only change I made to the file was to add in a description for DataSetExtensions so I'm not sure how the tabs got there. On Visual Studio it looks like all spaces, no tabs

Comment (nchikanov) [inline pkg/descriptions.json]: I'll try to change it back to how it was

File: src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj (hunk 2 above)
Comment (nchikanov) [inline src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj]: Working on adding compatibility right now :) Hopefully we'll fix it soon

File: src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj [hunk 3]
```diff
@@ -3,9 +3,12 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <ProjectReference Include="..\ref\System.Data.DataSetExtensions.csproj">
-      <SupportedFramework>netcoreapp2.0;net461;$(AllXamarinFrameworks)</SupportedFramework>
+      <SupportedFramework>netcoreapp2.0;net45;$(AllXamarinFrameworks)</SupportedFramework>
     </ProjectReference>
     <ProjectReference Include="..\src\System.Data.DataSetExtensions.csproj" />
+    <InboxOnTargetFramework Include="net45">
```
Comment (weshaggard) [inline src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj]: @ericstj do we have to also manually add the placeholder like you did in DirectoryServices? https://github.com/dotnet/corefx/blob/master/src/System.DirectoryServices/pkg/System.DirectoryServices.pkgproj#L13

Comment (nchikanov) [inline src/System.Data.DataSetExtensions/pkg/System.Data.DataSetExtensions.pkgproj]: @joperezr looked at this with me and adding this creates an unnecessary spot for net45, we already have it in the rooted lib

**Comment by weshaggard** [comment]
@dotnet/dnceng looks like we are having some issues with Jenkins machines. https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_all+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/2118/console

**Comment by nchikanov** [comment]
test Windows x64 Debug Build 
test UWP NETNative x86 Release Build 
test UWP CoreCLR x64 Debug Build 
test Packaging All Configurations x64 Debug Build 
test OSX x64 Debug Build 
test NETFX x86 Release Build

**Comment by danmosemsft** [comment]
@dotnet/dnceng `Error during cleanup: java.io.IOException: remote file operation failed: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest at hudson.remoting.Channel@1b919765:JNLP4-connect connection from 131.107.19.218/131.107.19.218:53282: hudson.remoting.ChannelClosedException: channel is already closed`

**Comment by mmitche** [comment]
@danmosemsft I rebooted the machine to see whether it would help...it's only happening on two machines

@dotnet-bot test OSX x64 Debug Build

**Comment by nchikanov** [comment]
@weshaggard Please add final review :)

File: pkg/descriptions.json [hunk 4]
```diff
@@ -441,6 +441,16 @@
     ]
   },
   {
+    "Name": "System.Data.DataSetExtensions",
+    "Description": "Provides extensions to form LINQ expressions and method queries against DataTable objects.",
+    "CommonTypes": [
+      "System.Data.DataTable",
```
Comment (ericstj) [inline pkg/descriptions.json]: I don't see DataTable in the reference assembly, nor SortExpressionBuilder. 
 https://github.com/nchikanov/corefx/blob/52c2e03a2fad1856e9f7b767ea4cdc2c3f79df23/src/System.Data.DataSetExtensions/ref/System.Data.DataSetExtensions.cs

**Pull request merged**